onnxruntime/core/providers/webnn/builders/impl/squeeze_unsqueeze_op_builder.cc

```diff
@@ -80,46 +80,79 @@ Status UnsqueezeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
     options.set("axes", emscripten::val::array(axes_data));
   }
 
-  emscripten::val output = model_builder.GetBuilder().call<emscripten::val>("unsqueeze", input, options);
+  emscripten::val output = emscripten::val::undefined();
+  if (node.OpType() == "Squeeze") {
+    output = model_builder.GetBuilder().call<emscripten::val>("squeeze", input, options);
+  } else {  // Unsqueeze
+    output = model_builder.GetBuilder().call<emscripten::val>("unsqueeze", input, options);
+  }
+
   model_builder.AddOperand(node.OutputDefs()[0]->Name(), std::move(output));
   return Status::OK();
 }
 
 // Operator support related.
 
-bool UnsqueezeOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers,
-                                           const Node& node,
-                                           const WebnnDeviceType /* device_type */,
-                                           const logging::Logger& logger) const {
+bool SqueezeUnsqueezeOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers,
+                                                  const Node& node,
+                                                  const WebnnDeviceType /* device_type */,
+                                                  const logging::Logger& logger) const {
   const auto& input_defs = node.InputDefs();
   std::vector<int64_t> input_shape;
   if (!GetShape(*input_defs[0], input_shape, logger))
     return false;
 
   // Unsqueeze opset 13 uses input 1 as axes, it needs to be an initializer.
   if (node.SinceVersion() >= 13) {
-    if (input_defs.size() < 2) {
-      LOGS(logger, ERROR) << "Input axes of Unsqueeze must be provided";
-      return false;
-    }
-    const auto& axes_name = input_defs[1]->Name();
-    if (!Contains(initializers, axes_name)) {
-      LOGS(logger, ERROR) << "Input axes of Unsqueeze must be known";
-      return false;
+    if (node.OpType() == "Squeeze") {
+      if (input_defs.size() < 1) {
+        LOGS(logger, ERROR) << "Squeeze has no input tensor";
+        return false;
+      }
+      // Axes for Squeeze is optional.
+      if (input_defs.size() > 1) {
+        const auto& axes_name = input_defs[1]->Name();
+        if (!Contains(initializers, axes_name)) {
+          LOGS(logger, ERROR) << "Input axes of Squeeze must be provided";
+          return false;
+        }
+      }
+    } else {  // Unsqueeze
+      if (input_defs.size() < 2) {
+        LOGS(logger, ERROR) << "Input axes of Unsqueeze must be provided";
+        return false;
+      }
+      const auto& axes_name = input_defs[1]->Name();
+      if (!Contains(initializers, axes_name)) {
+        LOGS(logger, ERROR) << "Input axes of Unsqueeze must be known";
+        return false;
+      }
```

Comment:
The code looks a little duplicated. Would it be equivalent to say

```
  if (node.SinceVersion() >= 13) {
    if (input_defs.size() < 1) {
      LOGS(logger, ERROR) << "Squeeze has no input tensor";
      return false;
    } else if (input_defs.size() > 1) {
      const auto& axes_name = input_defs[1]->Name();
      if (!Contains(initializers, axes_name)) {
        LOGS(logger, ERROR) << "Input axes of " << node.OpType() << " is not present and constant";
        return false;
      }
    } else if (node.OpType() == "Unsqueeze") {
        // The axes are optional for Squeeze, but not Unsqueeze.
        LOGS(logger, ERROR) << "Input axes of Unsqueeze must be provided";
    }
} else {
```

(I also updated the error message to "is not present and constant" because otherwise someone could provide an axis, but still get a confusing error message because it's not a constant)

Reply:
Thanks, updated.